.circleci/config.yml

```diff
@@ -1,9 +1,26 @@
-version: 2.0
-jobs:
-  build:
+version: 2.1
+executors:
+  docker-executor:
     docker:
       - image: outpostuniverse/ubuntu-18.04-gcc-sdl2-physfs-circleci
+  macos-executor:
+    macos:
+      xcode: "10.0.0"
+jobs:
+  build-linux:
+    executor: docker-executor
     steps:
-      - run: brew install physfs sdl2 sdl2_image sdl2_mixer sdl2_ttf glew
       - checkout
       - run: make -k
+  build-macos:
+    executor: macos-executor
+    steps:
+      - checkout
+      - run: brew install physfs sdl2 sdl2_image sdl2_mixer sdl2_ttf glew
+      - run: make -k
+workflows:
+  version: 2
+  linux_macos_build:
```

Comment:
I think I might just name this `build` as a way to reduce repetition. The extra detail doesn't really add anything, though the name could become misleading or add to maintenance if the steps below change. For instance, we might someday add a Windows build, or remove one of the existing Linux or MacOS builds. Ideally we shouldn't have to update the name here to keep it in sync.

Reply:
https://circleci.com/docs/2.0/configuration-reference/#workflows
https://circleci.com/docs/2.0/configuration-reference/#jobs

 - Workflow names must be unique.
 - Job names must be unique.
 - The key "build" has a special meaning.

Since this workflow encompassed building linux and then macos, that's what it was named. It's trivial to change it when needed, especially if you are already changing the file for something else.

Reply:
It's still an extra detail that has to be remembered. We should strive to reduce extra details like that.

The job name "build" is only special when not using workflows. Names must be unique (within a config.yml), but I think that's only with a section (workflow or job, not unique among both at the same time).

----

What about naming the workflow `build-and-test`? That's more descriptive of what it does (build and test), rather than how it does it (on linux and macos). It's the building and testing that's important, more so than what systems that's done on.

Reply:
Changed as requested. :)